source/modulo_controllers/src/ControllerInterface.cpp

```diff
@@ -52,6 +54,12 @@ rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn Contro
 
 rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
 ControllerInterface::on_configure(const rclcpp_lifecycle::State&) {
+  if (!on_init_called_) {
+    RCLCPP_ERROR(
+        get_node()->get_logger(),
+        "'ModuloControllerInterface::on_init()' has not been called, any derived class needs to do that");
```

Comment:
If we go with this approach, I would reword it:
```suggestion
        "The controller has not been properly initialized! Derived controller classes must call 'ControllerInterface::on_init()' during their own initialization before being configured.");
```

Also the class is now called `ControllerInterface`, not `ModuloControllerInterface`